ImputationPipeline/MergeSingleSampleMinimacVcfs.wdl

```diff
@@ -0,0 +1,434 @@
+                    }
+    }
+
+    Int disk_size_gb = 100
+    
+    command <<<
+        set -euo pipefail
+
+        cat <<'EOF' > script.py
+import hail as hl
+import pandas as pd
+
+# Calculate metrics
+hl.init(default_reference='GRCh37', idempotent=True)
+vcf = hl.import_vcf('~{imputed_vcf}', force_bgz=True)
+qc = hl.sample_qc(vcf)
+qc_pd = qc.cols().flatten() \
+    .rename({'sample_qc.' + col: col for col in list(qc['sample_qc'])}) \
+    .rename({'s': 'sample_id'}) \
+    .to_pandas()
+qc_pd.to_csv('~{output_basename}.qc_metrics.tsv', sep='\t', index=False, float_format='%.4f')
+EOF
+        python3 script.py
+    >>>
+
+    runtime {
+        docker: docker
+        disks: "local-disk " + disk_size_gb + " HDD"
+        memory: mem_gb + " GiB"
+        cpu: cpu
+        preemptible: preemptible
+    }
+
+    output {
+        File qc_metrics = "~{output_basename}.qc_metrics.tsv"
+    }
+}
+
+task GatherVcfs {
+
+  input {
+    Array[File] input_vcfs
+    String output_vcf_name
+    Int disk_size_gb = ceil(1.2 * size(input_vcfs, "GiB") + 50)
+    Int machine_mem_mb = 7000
+    String gatk_docker = "us.gcr.io/broad-gatk/gatk:4.5.0.0"
+    Int preemptible = 1
+  }
+
+  parameter_meta {
+    input_vcfs: {
+      localization_optional: true
+    }
+  }
+
+  command <<<
+    set -euo pipefail
+
+    # --ignore-safety-checks makes a big performance difference so we include it in our invocation.
+    # This argument disables expensive checks that the file headers contain the same set of
+    # genotyped samples and that files are in order by position of first record.
+    gatk --java-options "-Xms6000m -Xmx6500m" \
+      GatherVcfsCloud \
+      --ignore-safety-checks \
+      --gather-type BLOCK \
+      --input ~{sep=" --input " input_vcfs} \
+      --output ~{output_vcf_name}
+
+    tabix ~{output_vcf_name}
+  >>>
+
+  runtime {
+    memory: "~{machine_mem_mb} MiB"
+    cpu: "1"
+    bootDiskSizeGb: 15
+    disks: "local-disk " + disk_size_gb + " HDD"
+    preemptible: preemptible
+    docker: gatk_docker
+  }
+
+  output {
+    File output_vcf = "~{output_vcf_name}"
+    File output_vcf_index = "~{output_vcf_name}.tbi"
+  }
+}
+
+task MergeQCMetrics {
+    input {
+        Array[File] qc_metrics
+        String output_basename
+
+        Int disk_size_gb = ceil(2.2 * size(qc_metrics, "GiB") + 50)
+        Int mem_gb = 4
+        Int cpu = 2
+        Int preemptible = 1
+    }
+
+    command <<<
+        set -xeuo pipefail
+        
+
+        python3 <<EOF
+        import pandas as pd
+        qc_metrics = ['~{sep="', '" qc_metrics}']
+        merged_qc_metrics = pd.concat([pd.read_csv(qc_metric, sep='\t') for qc_metric in qc_metrics])
+        merged_qc_metrics.to_csv('~{output_basename}.qc_metrics.tsv', sep='\t')
+        EOF
+    >>>
+
+    output {
+        File merged_qc_metrics = "~{output_basename}.qc_metrics.tsv"
+    }
+
+    runtime {
+        docker: "us.gcr.io/broad-dsde-methods/samtools-suite:v1.1"
```

Comment:
I guess you only need `us.gcr.io/broad-dsde-methods/python-data-slim` here, but doesn't matter

Reply:
Sorry I clicked "add single comment" instead of making this part of a review